subsquid/src/pabloProcessor.ts

```diff
@@ -0,0 +1,299 @@
+
+        await ctx.store.save(pool);
+        await ctx.store.save(baseAsset);
+        await ctx.store.save(quoteAsset);
+        await ctx.store.save(tx);
+    } else {
+        throw new Error("Pool not found");
+    }
+}
+
+interface SwappedEvent {
+    poolId: bigint,
+    who: Uint8Array,
+    baseAsset: bigint,
+    quoteAsset: bigint,
+    baseAmount: bigint,
+    quoteAmount: bigint,
+    fee: bigint
+}
+
+function getSwappedEvent(event: PabloSwappedEvent): SwappedEvent {
+    if (event.isV2100) {
+        const {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee} = event.asV2100;
+        return {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee};
+    } else {
+        const {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee} = event.asLatest;
+        return {poolId, who, baseAsset, quoteAsset, baseAmount, quoteAmount, fee};
+    }
+}
+
+export async function processSwappedEvent(ctx: EventHandlerContext, event: PabloSwappedEvent) {
+    console.info('processing SwappedEvent', event);
+    const swappedEvt = getSwappedEvent(event);
+    const who = ss58.codec("picasso").encode(swappedEvt.who);
+    const pool = await get(ctx.store, PabloPool, swappedEvt.poolId.toString());
+    // only set values if the owner was missing, i.e a new pool
+    if (pool != undefined) {
+        const isReverse: boolean = pool.quoteAssetId != swappedEvt.quoteAsset;
+        const timestamp = BigInt(new Date().getTime());
+        pool.transactionCount += 1;
+        pool.calculatedTimestamp = timestamp;
+        pool.blockNumber = BigInt(ctx.block.height);
+        // find baseAsset: Following is only valid for dual asset pools
+        const baseAsset = pool.poolAssets
+            .find((asset) => asset.assetId != pool.quoteAssetId);
+        if (baseAsset == undefined) {
+            throw new Error('baseAsset not found');
+        }
+        // find quoteAsset
+        const quoteAsset = pool.poolAssets
+            .find((asset) => asset.assetId == pool.quoteAssetId);
+        if (quoteAsset == undefined) {
+            throw new Error('quoteAsset not found');
+        }
+        if (isReverse) {
+            console.debug('Reverse swap');
+            // volume
+            pool.totalVolume = Big(pool.totalVolume).add(Big(swappedEvt.baseAmount.toString())).toString();
+            baseAsset.totalVolume += swappedEvt.quoteAmount;
+            quoteAsset.totalVolume += swappedEvt.baseAmount;
+
+            // liquidity
+            pool.totalLiquidity = Big(pool.totalLiquidity)
+                // fees TODO is this correct?
+                .sub(Big(swappedEvt.fee.toString()))
+                .toString();
+            // for reverse exchange "default quote" (included as the base amount in the evt) amount leaves the pool
+            baseAsset.totalLiquidity += swappedEvt.quoteAmount;
+            quoteAsset.totalLiquidity -= swappedEvt.baseAmount;
+            quoteAsset.totalLiquidity -= swappedEvt.fee;
```

Comment:
I think here above calculation is only true for `StableSwap` as it applies fee on asset going out from pool. But in `Uniswap` we apply fee on asset coming in.  Perhpas we need to make them follow consistent way to apply fees.

Reply:
And most likely if we choose applying fee on asset going into pool then above computation will needed to be changed.

Reply:
Good point. I think the solution is to just include the `assetId` for fees on the event rather than change any fee calculations for pools (as it is better to just keep it to the spec of those pools for now for maintainability). I have a `TODO` below to correct the fee handling here, because I wasn't sure before.
But I will do that in a later PR.

Reply:
sure.